new_01-vim-beginner.tutor

```diff
@@ -0,0 +1,1234 @@
+
+  3. Oops, we haven't reached our target yet. Press «;» to repeat the search.
+
+  4. Now change the word.
+
+---> The brown fox jumps under the lazy dog. {expect:The brown fox jumps over the lazy dog.}
+---> The brown fox jumps over the lazy dog. {expect:The brown fox jumps over the lazy dog.}
+
+If we don't want to include the character we have searched for we can use the
+«t» motion. It stands for "'til" as in "until".
+
+  1. Place the cursor at the first character of the sequence to change.
+
+  2. Press «ct» to change 'til the period.
+
+---> This sentence was good until I lost it. {expect:This sentence is just right.}
+---> This sentence is just right. {expect:This sentence is just right.}
+
+We can also perform these motion in the other direction using their upper-case
+variants. This time the upper-case does not supercharge the motion, it does its
+opposite. That's also common convention. Try it out:
+
+  1. Place the cursor at the end of the sentence.
+
+  2. Press «cFu» to jump backwards and change the sentence.
+
+---> The brown fox jumps under the bus. {expect:The brown fox jumps over the lazy dog.}
+---> The brown fox jumps over the lazy dog. {expect:The brown fox jumps over the lazy dog.}
+
+It is not always a good idea to search for the character we want. If it's a very
+common character we will waste keystrokes skipping over the false positives.
+Often times it's better to jump to a nearby but less common character and then
+correct the position by one or two.
+
+# Lesson something.something: UNDO, REDO AND REPEAT
+
+** We can undo and redo changes. More imporantly though, we can repeat them **
+
+To undo a change press «u», press «C-R» to redo it again.
+
+  1. Make some change to the line to break it.
+
+  2. Undo your change by pressing «u».
+
+  3. You can re-apply the change pressing «C-R».
+
+---> Everything is great. {expect:Everything is great.}
+
+In the lessons above I have been metioning repeatability of actions and here is
+where it comes into play. We can repeat any last action using the period «.»
+operator.
+
+  1. Navigate to the first wrong word and change it with «ce».
+
+  2. After the change exit insert mode immediately and jump to the next word.
+
+  3. Now press «.» and see the change take effect immediately.
+
+---> Roses are blue, and tomatoes are blue as well. {expect:Roses are red, and tomatoes are red as well.}
+---> Roses are red, and tomatoes are red as well. {expect:Roses are red, and tomatoes are red as well.}
+
+An action consits of a count, an operator and a motion, you know that already.
+What I did not tell you is that everything you type in isert mode is part of
+the action as well. This means you can repeat entire phrases, sentences or even
+paragraphs. The only catch is that you cannot leave insert mode or it ends the
+action.
+
+The dot is actually a macro and always contains the last action. We will
+later learn how to define our own macros which are more powerful than that. We
```

Comment:
Right. I elaborated a bit more.

> The dot is actually a macro, a recorded command, and always contains the last action. We will later learn how to define our own macros which are more powerful than that and can contain any amout of actions. We will also learn how to transplant text properly.
> 
> The dot is still very powerful because it assigns itself automatically and you should use it whenever the possibility arises. Like most other operators it can be combined with a motion.